omnigibson/envs/vec_env_base.py

```diff
@@ -0,0 +1,51 @@
+import copy
+import time
+
+from tqdm import trange
+
+import omnigibson as og
+
+
+class VectorEnvironment:
+    def __init__(self, num_envs, config):
+        self.num_envs = num_envs
+        if og.sim is not None:
+            og.sim.stop()
+
+        # First we create the environments. We can't let DummyVecEnv do this for us because of the play call
+        # needing to happen before spaces are available for it to read things from.
+        self.envs = [
+            og.Environment(configs=copy.deepcopy(config), in_vec_env=True)
+            for _ in trange(num_envs, desc="Loading environments")
+        ]
+
+        # Play, and finish loading all the envs
+        og.sim.play()
+        for env in self.envs:
+            env.post_play_load()
+
+    def step(self, actions):
+        try:
+            observations, rewards, dones, infos = [], [], [], []
+            for i, action in enumerate(actions):
+                self.envs[i]._pre_step(action)
+            og.sim.step()
+            for i, action in enumerate(actions):
+                obs, reward, done, info = self.envs[i]._post_step(action)
+                observations.append(obs)
+                rewards.append(reward)
+                dones.append(done)
+                infos.append(info)
+            return observations, rewards, dones, infos
```

Comment:
mismatch in per-env return value (4-tuple) vs vec env return value (5-tuple)

Reply:
@hang-yin please fix this to always do 5-tuple. please update all .step callers to respect this too